build_tools/mako/benchmark_modules_on_android.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Benchmarks modules generated by compile_android_modules.py and writes
+# performance data to a text proto file, e.g., mako-phone-mako_tag-git_hash.
+#
+# The script benchmarks modules on 7-th big core, ie, run with `taskset 80`.
+
+import argparse
+import datetime
+import subprocess
+import os
+import re
+
+import configuration
+
+PATTERN = re.compile(r"BM_(\w+)(.+)/real_time(\s+) (?P<ms>[.0-9]+) ms")
+DEVICE_ROOT = "/data/local/tmp/benchmark_tmpdir"
+
+
+def parse_arguments():
+  parser = argparse.ArgumentParser()
+  parser.add_argument("--git_hash", default="UNKNOWN")
+  parser.add_argument("phone")
+  args = parser.parse_args()
+  return args
+
+
+def get_mako_sample(value, tag) -> str:
+  return f"""
+samples: {{
+  time: {value}
+  target: "{tag}"
+}}""".strip()
+
+
+def get_mako_metadata(git_hash, timestamp, benchmark_key) -> str:
+  return f"""
+metadata: {{
+  git_hash: "{git_hash}"
+  timestamp_ms: {timestamp}
+  benchmark_key: "{benchmark_key}"
+}}
+""".strip()
+
+
+def benchmark(module_name, flagfile_name, target) -> str:
+  samples = []
+  driver = target.get_driver()
+  cmd = [
+      "adb", "shell", "LD_LIBRARY_PATH=/data/local/tmp", "taskset", "80",
+      f"{DEVICE_ROOT}/iree-benchmark-module",
+      f"--flagfile={DEVICE_ROOT}/{flagfile_name}",
+      f"--module_file={DEVICE_ROOT}/{module_name}", f"--driver={driver}",
+      "--benchmark_repetitions=10"
+  ] + target.runtime_flags
+  print(f"Running cmd: {' '.join(cmd)}")
+  output = subprocess.run(cmd, check=True,
+                          capture_output=True).stdout.strip().decode()
```

Comment:
Can also set `universal_newlines` or `encoding` in `subprocess.run` (yes the API on that is bonkers).


```suggestion
  output = subprocess.run(cmd,
                          check=True,
                          capture_output=True,
                          universal_newlines=True).stdout.strip()
```

```suggestion
  output = subprocess.run(cmd,
                          check=True,
                          capture_output=True,
                          universal_newlines=True).stdout
```

Or is strip actually necessary here, since you then filter by pattern?